aptos-move/framework/src/natives/signature.rs

```diff
@@ -18,7 +19,123 @@ use move_deps::{
 use smallvec::smallvec;
 use std::{collections::VecDeque, convert::TryFrom};
 
-pub fn native_bls12381_public_key_validation(
+/// Returns the equivalent of a Move std::option::none() natively in Rust.
+/// TODO: vector_for_testing_only is not an API we conceptually support and misusing it could cause the VM to crash.
+fn none_option() -> Value {
+    Value::struct_(Struct::pack(std::iter::once(
+        Value::vector_for_testing_only(std::iter::empty()),
+    )))
+}
+
+/// Returns the equivalent of a Move std::option<vector<u8>>::some(v) natively in Rust.
+/// TODO: vector_for_testing_only is not an API we conceptually support and misusing it could cause the VM to crash.
+fn some_option(v: Vec<u8>) -> Value {
+    let vv = Value::vector_u8(v.into_iter());
+    Value::struct_(Struct::pack(std::iter::once(
+        Value::vector_for_testing_only(std::iter::once(vv)),
+    )))
+}
+
+/// Used to pop a Vec<Vec<u8>> argument off the stack.
+macro_rules! pop_vec_arg {
+    ($arguments:ident, $t:ty) => {{
+        // Replicating the code from pop_arg! here
+        use move_deps::move_vm_types::natives::function::{PartialVMError, StatusCode};
+        let value_vec = match $arguments.pop_back().map(|v| v.value_as::<Vec<Value>>()) {
```

Comment:
@davidiw you asked me to do a demo PR for this failing a few days ago. What was messing was the implementation of this cast to `Vec<Value>`, which @vgao1996 added in [this PR](https://github.com/move-language/move/pull/287)

Reply:
I don't think this macro belong to this file?

Reply:
Ideally we should add this to move-vm-types, but I probably wouldn't bother with it for now.

Reply:
That would be in the upstream Move repos, right? I'll leave that up to you and @wrwg. If there's a better place for it inside `aptos-move/framework` let me know.